tests/sparseml/core/lifecycle/test_event.py

```diff
@@ -0,0 +1,565 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from functools import partial
+
+import pytest
+
+from sparseml.core.event import Event, EventType
+from sparseml.core.lifecycle.event import (
+    CallbacksEventLifecycle,
+    EventLifecycle,
+    WrappedOptimEventLifecycle,
+)
+
+
+def test_event_lifecycle_can_not_be_instantiated():
+    with pytest.raises(TypeError):
+        EventLifecycle(type_first=EventType.BATCH_START, start=Event())
```

Comment:
Could you add a short comment explaining why this should fail? (I assume its because its an abstract class but I had to go into the source code to check that)

Reply:
Done!